packages/core/echo/echo-schema/src/query/filter.ts

```diff
@@ -214,25 +222,28 @@ const filterMatchInner = (filter: Filter, object: EchoObject): boolean => {
     return false;
   }
 
-  // TODO(burdon): Should match by default?
-  let match = true;
-
   if (filter.type) {
     if (!isTypedObject(object)) {
       return false;
     }
 
-    const type = object[base]._getType();
-    if (!type) {
-      return false;
-    }
+    // Separate branch for objects with dynamic schema and typename filters.
```

Comment:
Can you fix and remove: `static _typename(typename: string)` (incl. all usages).